presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java

```diff
@@ -87,6 +88,17 @@ public CassandraPartitionResult getPartitions(ConnectorTableHandle tableHandle,
                 @SuppressWarnings({"rawtypes", "unchecked"})
                 List<ColumnHandle> partitionColumns = (List) partitionKeys;
                 remainingTupleDomain = TupleDomain.withColumnDomains(Maps.filterKeys(tupleDomain.getDomains().get(), not(in(partitionColumns))));
+                List<CassandraColumnHandle> clusteringColumns = table.getClusteringKeyColumns();
```

Comment:
I don't think you need to prune clustering predicates at this step. It must be prunned in a place where you actually build the clustering predicate. And it must be prunned from the `unenforced predicates` returned from the partition manager.